### PR TITLE
harden(nf-prompt): fix fan-out over-dispatch on a Claude-only quorum budget

### PR DESCRIPTION
**Pull request by jobordu**
## What

`/nf:harden` on `hooks/nf-prompt.js` — the **input-side enforcement** that requires the quorum for planning commands and sets the fan-out fleet size. It's the companion to nf-stop's output gate (#279): nf-stop blocks delivery without consensus; nf-prompt *requires* the quorum and sizes the fleet. Converged over 2 rounds.

## 🔴 The gap — SC-4 fan-out over-dispatch / header-body contradiction

The "graceful fallback" (SC-4) force-restored one external slot whenever the dispatch list was empty — ignoring that `externalSlotCap === 0` (`fanOutCount === 1`) is the **intentional "Claude is the whole quorum"** budget:

- **`max_quorum_size=1`**: header said `QUORUM REQUIRED (--n 1)` but the body dispatched 1 slot-worker Task → fleet = Claude + 1 = **2 > maxSize=1** (out-of-range over-dispatch).
- **default `maxSize=3` + `risk_level=routine`** (`ceil(3/3)=1` → 0 external): the header announced `→ 0 external slots` yet the body listed 1 Task — a **self-contradiction**, reachable on *every* routine/low planning task.

## Fix

A 0-external budget (`externalSlotCap <= 0`) now routes to **SOLO MODE** before the dispatch path — the same `NF_SOLO_MODE` marker the empty-roster and explicit `--n 1` paths already emit, which the Stop hook recognises so it does **not** floor-block on 0 external voters. (Rendering "QUORUM REQUIRED + 0 slots" would instead *deadlock* the gate at `min_live_voters=2`.) SC-4 also gains a defensive `externalSlotCap > 0` guard so it can never restore a slot against an intentional 0 budget.

## Verification (2 rounds, every fix red-proven)

- **Round 1**: SC-4 gap fixed. Other surfaces confirmed clean — trigger anchoring (no false-trigger, no bypass), `parseQuorumSizeFlag`/`mapRiskLevelToCount` stay in `[1..maxSize]`, corrupt scoreboard/providers fail-open, and the breaker/queued-task/priority-ordering/output-contract/fail-open paths (+6 tests).
- **Round 2**: CONVERGED. A 13-cell `maxSize × risk × --n` matrix confirms `externals == max(0, fanOut-1)`, solo iff `fanOut===1`, never `> maxSize-1`; the critical boundary `externalSlotCap === 1` is **not** swallowed (still dispatches 1); SC-4 restore on a `>0` budget keeps header/body consistent (+5 tests).

**nf-prompt 56/56; lint OK; source↔dist in sync.**

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved quorum handling so solo-mode now correctly skips external task dispatches when no external slots are available.
  * Fixed an over-dispatch issue that could restore a slot when the external budget should remain zero.
  * Made fallback behavior more consistent for edge cases, including malformed or missing state.

* **Tests**
  * Added broader coverage for quorum priority, solo-mode behavior, dispatch limits, and fail-open handling in unusual scenarios.

### CODE REVIEW

File: hooks/nf-prompt.js [hunk 1]
```diff
@@ -602,11 +602,27 @@ process.stdin.on('end', () => {
...
+      if (orderedSlots.length === 0 || externalSlotCap <= 0) {
+        if (orderedSlots.length === 0) {
+          // Fail-open to solo mode: Claude is the only quorum participant
+          process.stderr.write('[nf-dispatch] WARNING: no external agents in roster — falling back to solo quorum\n');
+          instructions = `<!-- NF_SOLO_MODE -->\nSOLO MODE ACTIVE (empty roster): No external agents configured in providers.json or quorum_active. Claude's vote is the quorum. Write <!-- GSD_DECISION --> in your final output. The Stop hook is informed.\n\nTo add agents, run /nf:mcp-setup or edit ~/.claude/nf.json quorum_active.\n`;
+        } else {
+          process.stderr.write('[nf-dispatch] solo quorum — fan-out budget is 1 (Claude only), dispatching 0 external slots\n');
+          instructions = `<!-- NF_SOLO_MODE -->\nSOLO MODE ACTIVE (fan-out budget = 1): For this task the quorum size is 1 (from risk_level / max_quorum_size), so Claude is the entire quorum. Skip ALL external slot-worker Task dispatches. Claude's vote is the quorum. Write <!-- GSD_DECISION --> in your final output. The Stop hook is informed.\n\n`;
```
[the remark this answers is not in the document]
Comment (jobordu) [reply hooks/nf-prompt.js]: Fixed in the latest commit — reworded to reference the real mechanism (risk_level, the configured quorum size quorum.maxSize, and/or the --n cap) instead of the unused 'max_quorum_size' name.

File: hooks/nf-prompt.test.js [hunk 2]
```diff
@@ -1397,3 +1597,431 @@ test('ADVERSARIAL: parseQuorumSizeFlag drops trailing non-digit characters like
...
+  } finally {
+    fs.rmSync(tempDir, { recursive: true, force: true });
+  }
+});
+
+// ══════════════════════════════════════════════════════════════════════════════
+// ── ADVERSARIAL: pure-helper boundary invariants (no subprocess) ─────────────
+// ══════════════════════════════════════════════════════════════════════════════
+
+const nfPrompt = require('./nf-prompt.js');
```
[the remark this answers is not in the document]
Comment (jobordu) [reply hooks/nf-prompt.test.js:1710]: This is a false positive — lint:isolation explicitly SKIPS test files (scripts/lint-isolation.js line 121: it 'continue's on *.test.js/.cjs/.mjs), and the rule targets require('./bin/...') paths, not a colocated hook-under-test. This require('./nf-prompt.js') matches the established convention (e.g. hooks/nf-session-start.test.js:427 requires its own hook the same way) and npm run lint:isolation passes. No change needed.
